fs2-aws/src/main/scala/fs2/aws/s3/s3.scala

```diff
@@ -57,7 +57,7 @@ package object s3 {
                   .withBucketName(bucket)
                   .withKey(key)
                   .withUploadId(uploadId)
-                  .withPartNumber(i.toInt)
+                  .withPartNumber(i.toInt + 1)
```

Comment:
I think it's better to alter the source of the problem, in this case the zip. 
Instead of zipWithIndex we can do `.zip(Stream.iterate(1)(_ + 1))`